### PR TITLE
TD-883 Add CSAT Rating mid-conversation (Before it has been resolved)

### PR DESCRIPTION
**Pull request by Prochnost55**
### What do you want to achieve?
- To add a functionality for rating the conversation before it gets resolved.

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [x] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

- Running on localhost
Screenshot 2020-12-28 at 6 04 26 PM

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/css/app/mck-sidebox-1.0.css [hunk 1]
```diff
@@ -5413,6 +5416,51 @@ div#mck-waiting-queue {
 	padding: 12px 5px 13px 5px;
 	border-top: 1px dashed #CAD3E3;
 }
+.csatModule{
```
Comment (devashishmamgain) [inline webplugin/css/app/mck-sidebox-1.0.css:5419]: use proper naming convention

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 2]
```diff
@@ -5438,7 +5449,27 @@ var KM_ATTACHMENT_V2_SUPPORTED_MIME_TYPES = ["application","text","image"];
                 }];
 
                 append ? $applozic.tmpl("messageTemplate", msgList).appendTo("#mck-message-cell .mck-message-inner") : $applozic.tmpl("messageTemplate", msgList).prependTo("#mck-message-cell .mck-message-inner");
-                       
+                
+                if (msg.contentType = KommunicateConstants.MESSAGE_CONTENT_TYPE.NOTIFY_MESSAGE) {
+                    if (msg.metadata && msg.metadata.feedback) {
+                        let userFeedback = JSON.parse(msg.metadata.feedback);
+                        let ratingSmileSVG = kommunicateCommons.getRatingSmilies(
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:5455]: Use var in chat widget code

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 3]
```diff
@@ -7091,7 +7122,9 @@ var KM_ATTACHMENT_V2_SUPPORTED_MIME_TYPES = ["application","text","image"];
                                 }
                             }
                         }
-                    } else if (messageType === "APPLOZIC_02" && !message.contentType == 102) {
+                    } else if (
+                        messageType === "APPLOZIC_02" && !message.contentType == 102 || 
+                        messageType === "APPLOZIC_02" && message.metadata.feedback && !kommunicate._globals.isConvJustResolved) {
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:7126]: If conversion is resolved then below code is not required?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/kommunicate.js [hunk 4]
```diff
@@ -518,9 +518,6 @@ $applozic.extend(true,Kommunicate,{
     // check if the message needs to be processed by addMessage
     visibleMessage: function(msg){
         if(!msg) return false;
-        if(msg.metadata && msg.metadata.feedback){
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicate.js:521]: what was the purpose of this code which is removed? is it not required now?

File: webplugin/js/app/kommunicate-ui.js [hunk 5]
```diff
@@ -440,62 +446,148 @@ setAvailabilityStatus : function (status){
...
                 */
                 if (feedback && feedback.rating) {
-                    if(feedback.comments.length > 0){ // if comments are there in feedback 
-                        kommunicateCommons.modifyClassList( {id : ["csat-3","mck-rated"]}, "", "n-vis");
-                        document.getElementById('csat-3').innerHTML = '\"' + feedback.comments[0] + '\"';
-                    } else { // only rating via emoticons
-                        kommunicateCommons.modifyClassList( {id : ["csat-2","mck-rated"]}, "", "n-vis");
-                    }
-                    document.getElementById('mck-rating-container').innerHTML = kommunicateCommons.getRatingSmilies(feedback.rating);
+                    // if(feedback.comments.length > 0){ // if comments are there in feedback
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicate-ui.js:515]: is this logic moved somewhere else?

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 6]
```diff
@@ -9819,6 +9851,11 @@ var KM_ATTACHMENT_V2_SUPPORTED_MIME_TYPES = ["application","text","image"];
                                         KommunicateUI.showClosedConversationBanner(true);
                                     }, MCK_BOT_MESSAGE_DELAY);
                                 } else {
+                                    if (kommunicate._globals.isConvJustResolved) {
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:9854]: can this be replaced with:
kommunicate._globals.isConvJustResolved = !kommunicate._globals.isConvJustResolved

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/kommunicate-ui.js (hunk 5 above)
Comment (Prochnost55) [reply webplugin/js/app/kommunicate-ui.js:515]: This portion of code is to be ignored so made the suitable changes

File: webplugin/js/app/kommunicate.js (hunk 4 above)
Comment (Prochnost55) [reply webplugin/js/app/kommunicate.js:521]: Previously, whenever there was a message containing feedback it was ignored.

Now it is not ignored as a user may give multiple feedbacks (CSAT ratings) and is shown along with the messages as is shown in the dashboard. 

Screenshot 2020-12-29 at 4 50 35 PM

**Review by devashishmamgain — APPROVED** [review]

**Review by ab14bhardwaj — APPROVED** [review]

**Review by ab14bhardwaj — DISMISSED** [review]

File: webplugin/js/app/kommunicate-ui.js [hunk 7]
```diff
@@ -11,6 +11,8 @@ KommunicateUI={
     leadCollectionEnabledOnWelcomeMessage:false,
     anonymousUser:false,
     showResolvedConversations: false,
+    CSATtriggeredByCustomer: false,
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicate-ui.js:14]: naming convention

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

**Review by devashishmamgain — APPROVED** [review]

**Review by ab14bhardwaj — APPROVED** [review]

**Comment by Prochnost55** [comment]
changed branch name as requested by vipin sir. Bash doesn't support ()

**Pull request closed without merging**